### PR TITLE
[ISSUE #4658] remove redundant logic

### LINKED ISSUE #4658: remove redundant logic in checkTransactionState
- in `DefaultMQProducerImpl#checkTransactionState`, there is redundant if logic, need to be removed

image

- fix getacl config desc

### PR DESCRIPTION
**Pull request by zhiliatom**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

close #4658

### CODE REVIEW

File: client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java [hunk 1]
```diff
@@ -348,11 +348,9 @@ public void run() {
                     try {
                         if (transactionCheckListener != null) {
                             localTransactionState = transactionCheckListener.checkLocalTransactionState(message);
-                        } else if (transactionListener != null) {
+                        } else {
                             log.debug("Used new check API in transaction message");
```
Comment (aaron-ai) [inline client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java:352]:
```suggestion
                            log.warn("transactionListener is null, producerGroup={}", group);
```

File: client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java [hunk 2]
```diff
@@ -348,11 +348,9 @@ public void run() {
                     try {
                         if (transactionCheckListener != null) {
                             localTransactionState = transactionCheckListener.checkLocalTransactionState(message);
-                        } else if (transactionListener != null) {
-                            log.debug("Used new check API in transaction message");
-                            localTransactionState = transactionListener.checkLocalTransaction(message);
                         } else {
-                            log.warn("CheckTransactionState, pick transactionListener by group[{}] failed", group);
+                            log.warn("transactionListener is null, producerGroup={}", group);
```
Comment (lizhimins) [inline client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java:352]: The log information maybe not correct. The transactionCheckListener is empty. It‘s new API

Comment (lizhimins) [inline client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java:352]:
```suggestion
                            log.warn("TransactionCheckListener is null, used new check API, producerGroup={}", group);
```

File: client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java [hunk 3]
```diff
@@ -348,11 +348,9 @@ public void run() {
                     try {
                         if (transactionCheckListener != null) {
                             localTransactionState = transactionCheckListener.checkLocalTransactionState(message);
-                        } else if (transactionListener != null) {
-                            log.debug("Used new check API in transaction message");
-                            localTransactionState = transactionListener.checkLocalTransaction(message);
                         } else {
-                            log.warn("CheckTransactionState, pick transactionListener by group[{}] failed", group);
+                            log.warn("TransactionCheckListener is null, used new check API, producerGroup={}", group);
```
Comment (aaron-ai) [inline client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java:352]:
```suggestion
                            log.debug("TransactionCheckListener is null, used new check API, producerGroup={}", group);
```

**Review by aaron-ai — APPROVED** [review]

**Pull request merged**